src/lib/profiles/software-update/SoftwareUpdateProfile.h

```diff
@@ -59,357 +59,389 @@ namespace nl {
-    uint16_t packedLength();
-    static WEAVE_ERROR parse(MessageIterator&, IntegrityTypeList&);
+    WEAVE_ERROR pack(MessageIterator &);
+    static WEAVE_ERROR parse(MessageIterator &, IntegrityTypeList &);
     // comparison
-    bool operator == (const IntegrityTypeList&) const;
-    // this is just a short list with specified length. if length is
-    // 0 this indicates that the list is undefined since, to be useful,
-    // one of these things has to caontain at least one element.
-    uint8_t theLength;
-    uint8_t theList[3];
-  };
-  /*
-   * an auxiliary class that holds a list of update schemes as part of
-   * the image query. these are pretty simple and are essenitlly wrappers
-   * around a 4-element array of values drawn from the above enumeration.
-   */
-  class NL_DLL_EXPORT UpdateSchemeList {
-  public:
+    bool operator ==(const IntegrityTypeList &) const;
+    uint8_t theLength;  /**< Length of the supported element list.  Length of 0 indicates an empty list */
+    uint8_t theList[3]; /**< Container holding supported integrity types.  It is sized equal to the number of elements in @ref
+                           IntegrityTypes */
+};
+
+/**
+ * An auxiliary class to hold a list of update schemes as a part of the image query.
+ *
+ * A simple wrapper, sized to hold any subset of possible update schemes. In
+ * order to accomplish this task, its size is equal to the number of elements in
+ * the @ref UpdateSchemes is used to generate the list of supported
+ * update schemes in the ImageQuery message.
+ */
+class NL_DLL_EXPORT UpdateSchemeList
+{
+public:
     // constructor
     UpdateSchemeList();
     // initializer
-    WEAVE_ERROR init(uint8_t, uint8_t*);
+    WEAVE_ERROR init(uint8_t, uint8_t *);
     // packing and parsing
-    WEAVE_ERROR pack(MessageIterator&);
-    uint16_t packedLength();
-    static WEAVE_ERROR parse(MessageIterator&, UpdateSchemeList&);
+    WEAVE_ERROR pack(MessageIterator &);
+    static WEAVE_ERROR parse(MessageIterator &, UpdateSchemeList &);
     // comparison
-    bool operator == (const UpdateSchemeList&) const;
-    // data members
-    // this is just a short list with specified length. if length is
-    // 0 this indicates that the list is undefined since, to be useful,
-    // one of these things has to caontain at least one element.
-    uint8_t theLength;
-    uint8_t theList[4];
-  };
-  /*
-   * an auxiliary class that represents a product specification, which
-   * for our purposes will consist of:
-   * - a 16-bit vendor identifier
-   * - a 16-bit product identifier
-   * - a 16-bit porduct revision number
-   */
-  class NL_DLL_EXPORT ProductSpec {
-  public:
+    bool operator ==(const UpdateSchemeList &) const;
+    uint8_t theLength;  /**< Length of the supported element list.  Length of 0 indicates an empty list */
+    uint8_t theList[4]; /**< Container holding supported update schemes.  It is sized equal to the number of elements in @ref
```

Comment:
4: magic number -> change to a mnemonic and use here and in the source code?

Reply:
done